project/src/test/java/com/google/sps/data/DataAccessorTest.java

```diff
@@ -0,0 +1,110 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class DataAccessorTest {
+
+  // A helper that enables us to test datastore locally.
+  // Has to be set up and teared down for each test.
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  // A datastore service instance, that shall be created locally
+  private final DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+
+  // The tested dataaccessor, which will be initialized with a datastore service instance.
+  private DataAccessor dataAccessor;
+
+  @Before
+  public void setUp() {
+      helper.setUp();
+      dataAccessor = new DataAccessor(datastoreService);
+  }
+
+  @After
+  public void tearDown() {
+      helper.tearDown();
+  }
+
+  @Test
+  public void isRegisteredId_registered_true() {
+    String userId = "12345";
+    Entity userEntity = new Entity(dataAccessor.userEntityName, userId);
+    datastoreService.put(userEntity);
+
+    assertTrue(dataAccessor.isRegisteredId(userId));
+  }
+
+  @Test
+  public void isRegisteredId_notRegistered_false() {
+    String registeredUserId = "12345";
+    String unRegisteredUserId = "54321";
+    Entity userEntity = new Entity(dataAccessor.userEntityName, registeredUserId);
+    datastoreService.put(userEntity);
+
+    assertFalse(dataAccessor.isRegisteredId(unRegisteredUserId));
+  }
+
+  @Test
+  public void registerUserId_validUser_registerUser() {
+    String userId = "12345";
+
+    dataAccessor.registerUserId(userId);
+    PreparedQuery results = createPreparedQueryByUserId(userId);
+
+    assertEquals(results.asList(FetchOptions.Builder.withDefaults()).size(), 1);
+  }
+
+  private PreparedQuery createPreparedQueryByUserId(String userId) {
```

Comment:
This function seems exactly like the isRegistered function (added the assertion that comes after this on the `registerUserId_validUser_registerUser` test).
So this makes me wonder if this "copy" of the function here is really needed, and if there is in general anything to do with the dependency of the registerUserId testing on the isRegistered tests.
I am just pointing this out, I am not really sure what is best to do in this situation.

Reply:
Right. I thought about it, and so far I didn't find a better way to test if the entity was added. 
What I said to myself (which made me feel better about this) is that these tests check the desired behaviour, and so if someone decides to change the functions' implementation, the functions still must work as we originally expected, even if the implementation will be different. I'm not sure if this helps haha. LMK if you think of an improvement or a different way to do that.

Reply:
That makes sense, and I don't have a better solution. I guess there is no way to avoid the dependency between `registerUserId` and `isRegistered` when testing.